File: src/main/java/com/uiptv/server/api/json/HttpConfigJsonServer.java
package com.uiptv.server.api.json;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.uiptv.service.ConfigurationService;
import org.json.JSONObject;

import java.io.IOException;

import static com.uiptv.util.ServerUtils.generateJsonResponse;

public class HttpConfigJsonServer implements HttpHandler {
    @Override
    public void handle(HttpExchange exchange) throws IOException {
        JSONObject response = new JSONObject();
        response.put("enableThumbnails", ConfigurationService.getInstance().read().isEnableThumbnails());
        generateJsonResponse(exchange, response.toString());
    }
}

File: server/src/main/kotlin/com/uiptv/server/api/json/HttpConfigJsonServer.kt
package com.uiptv.server.api.json

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import com.uiptv.service.ConfigurationService
import com.uiptv.util.ServerUtils.generateJsonResponse
import org.json.JSONObject
import java.io.IOException

class HttpConfigJsonServer : HttpHandler {
    @Throws(IOException::class)
    override fun handle(exchange: HttpExchange) {
        val response = JSONObject()
        response.put("enableThumbnails", ConfigurationService.getInstance().read().enableThumbnails)
        generateJsonResponse(exchange, response.toString())
    }
}
